integration-tests/it-consumer/src/main/java/org/apache/servicecomb/it/testcase/base/TestRestServerConfig.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.it.testcase.base;
+
+import org.apache.servicecomb.it.Consumers;
+import org.apache.servicecomb.it.junit.ITJUnitUtils;
+import org.apache.servicecomb.swagger.invocation.exception.InvocationException;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TestRestServerConfig {
+  interface RestServerConfigSchemaIntf {
+    String testMaxInitialLineLength(String q);
+  }
+
+  static Consumers<RestServerConfigSchemaIntf> consumers = new Consumers<>("restServerConfig",
+      RestServerConfigSchemaIntf.class);
+
+  @BeforeClass
+  public static void classSetup() {
+    consumers.init(ITJUnitUtils.getTransport());
+  }
+
+  @Test
+  public void testMaxInitialLineLength_underLimit() {
+    String querySegment = "1234567890";
+    StringBuilder q = new StringBuilder();
+    while (q.length() < 4900) {
+      q.append(querySegment);
+    }
+    String result = consumers.getIntf().testMaxInitialLineLength(q.toString());
+    Assert.assertEquals("OK", result);
+  }
+
+  @Test
+  public void testMaxInitialLineLength_exceedLimit() {
```

Comment:
test 4999 and 5000?

Reply:
Fixed, and there is conflict between this pr and master, so I rebase this branch.